src/java/org/apache/cassandra/index/sai/disk/OnDiskKeyProducer.java

```diff
@@ -75,25 +71,12 @@ private long getKeyOffset(long segmentRowId)
             return NO_OFFSET;
         }
 
-        // Due to ZCS, index files may still contain partition offsets that are not part of partial SSTable.
-        if (offset > maxPartitionOffset)
-        {
-            exhausted = true;
-            return NO_OFFSET;
-        }
-
         // Catalog the last offset if it's valid:
         lastOffset = offset;
 
         return offset;
     }
 
-    public boolean isExhausted()
```

Comment:
Should this be used by `PostingListTokenFlow#exhausted`? Like 
```
return needsSkipping && skipToToken > getMaximum() || producer.isExhausted();
```

Reply:
maybe not if offset never exceeds maxPartitionOffset?

Reply:
yeah, offset never exceeds maxPartitionOffset.